src/module/io.wren

```diff
@@ -107,6 +108,44 @@ foreign class File {
     return Scheduler.await_ { realPath_(path, Fiber.current) }
   }
 
+  static dirname(path) {
+    var separator = Platform.directorySeparator
+    if (!path.contains(separator)) return "."
+
+    path = path.trimEnd(separator)
+    if (path == "") return separator
+    if (!path.contains(separator)) return "."
+
+    var i = lastIndexOf(path, separator)
+    var dirname = path[0..i].trimEnd(separator)
+    return dirname == "" ? separator : dirname
+  }
+
+  static lastIndexOf(string, char) {
```

Comment:
This belongs in essentials, eventually.

Reply:
Agree, but currently it's problematic for Wren Console to have dependencies on Essentials since theoretically it should be compilable without as well... (and we're still not sure how closely we'll work with the Wren CLI project - where we can't assume Essentials is present).

Reply:
This also simplifies testing since it removes the conditionals...

Reply:
Could we move this to a `CoreExtra` class/module (open to naming suggestions) for keeping track of such utilities that can't [yet] be moved to essentials?  There is an issue in Wren repo for adding this to core Wren, but no progress on it.